tests/src/helpers.c

```diff
@@ -244,3 +244,16 @@ void mbedtls_param_failed( const char *failure_condition,
     }
 }
 #endif /* MBEDTLS_CHECK_PARAMS */
+
+#if defined(MBEDTLS_TEST_HOOKS)
+void mbedtls_test_err_add_check( int high, int low,
+                                 const char *file, int line )
+{
+    if ( high > -0x1000 || low < -0x007F )
```

Comment:
These checks are incomplete.

* `high` and `low` should be negative or zero. And I'm not even sure about allowing zero.
* `high` should not contain a low-level error code already.

Reply:
I'm not sure I completely agree with that statement.

- `high` is always negative and therefore cannot be zero. `low` on the other hand can be positive and hence can be zero, I am happy to add this check. I completely agree that zero should not be allowed as it would not make sense to add a success to an error (that is still an error).

- "high should not contain a low-level error code already."
I'm not entirely sure what you mean by this. The check throws an error if `high` is negatively smaller than a high error code and thus any low error code will be found to be incorrect and therefore the check will throw an error.

Reply:
`high=-0x2104` should be not allowed. That's not a high-level error code, it's already a combination of a high-level error code with a low-level error code.

Reply:
Ah okay, thanks for the clarification. I will make the changes now.

Reply:
Just so you are aware in my next push I have allowed zero to pass the check as I found a case (`rsa.c:1282`) where `ret` can either be zero or an error. This could be something to change in the future but I don't think it makes enough of a difference for us to care about now.

Reply:
Done.